packages/attune/src/server/repo/index/sign.rs

```diff
@@ -679,97 +708,46 @@ pub async fn handler(
-            // component-package, we never delete the package row, and
-            // successful index generation implies the package row exists.
-            let package =
-                Package::query_from_meta(&mut tx, &tenant_id, name, version, architecture)
-                    .await
-                    .ok_or(ErrorResponse::new(
-                        StatusCode::NOT_FOUND,
-                        "PACKAGE_NOT_FOUND".to_string(),
-                        "package not found".to_string(),
-                    ))?;
-
-            let pool_filename = package.pool_filename_in_component(&req.change.component);
-
-            // Determine whether there exist other component-packages with the
-            // same filename.
-            let remaining_component_packages = sqlx::query!(
-                r#"
-                SELECT COUNT(*) AS "count!: i64"
-                FROM
-                    debian_repository_package
-                    JOIN debian_repository_component_package ON debian_repository_package.id = debian_repository_component_package.package_id
-                WHERE
-                    debian_repository_package.tenant_id = $1
-                    AND debian_repository_package.package = $2
-                    AND debian_repository_package.version = $3
-                    AND debian_repository_package.architecture = $4::debian_repository_architecture
-                    AND debian_repository_component_package.filename = $5
-                "#,
-                tenant_id.0,
-                name,
-                version,
-                architecture as _,
-                pool_filename,
-            )
-            .fetch_one(&mut *tx)
-            .await
-            .unwrap();
-
-            tx.commit().await.unwrap();
-
+        PackageChangeAction::Remove { .. } => {
             // Delete the pool file from S3 if it's fully orphaned.
-            if remaining_component_packages.count == 0 {
-                let key = format!("{}/{}", repo.s3_prefix, pool_filename);
-                tracing::debug!(?key, "deleting pool file");
-                state
-                    .s3
-                    .delete_object()
+            let key = format!("{}/{}", repo.s3_prefix, result.changed_package.filename);
+            debug!(?key, "delete pool file from S3");
+            if result.orphaned_pool_filename {
+                s3.delete_object()
                     .bucket(&repo.s3_bucket)
-                    .key(key)
+                    .key(format!(
+                        "{}/{}",
+                        repo.s3_prefix, result.changed_package.filename
+                    ))
                     .send()
                     .await
                     .unwrap();
             }
         }
     }
 
-    // Upload the updated package index files to standard path and all by-hash paths concurrently.
+    // Upload the updated package index files to standard path and all by-hash
+    // paths concurrently.
```

Comment:
happy to set my editor to wrap at the same width as you if you would like, what width do you have it set to?

Reply:
I'm wrapping at 80 but this isn't super important to me. I like 80 because it wraps nicely on mobile, but also I think the _right_ way to enforce something like this is a linter. `rustfmt` has [an option](https://rust-lang.github.io/rustfmt/?version=v1.8.0&search=#comment_width) for this, but it's unstable right now. I might just add it to the next PR.